controllers/drplacementcontrol_controller.go

```diff
@@ -1264,9 +1264,53 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("VRG adoption in progress")
 	}
 
-	// delete manifestworks (VRGs)
+	// delete manifestworks (VRGs and NSs)
 	for _, drClusterName := range rmnutil.DRPolicyClusterNames(drPolicy) {
-		err := mwu.DeleteManifestWorksForCluster(drClusterName)
+		// delete VRG manifestwork
+		mwName := mwu.BuildManifestWorkName(rmnutil.MWTypeVRG)
+		mwNamespace := drClusterName
+
+		err = mwu.DeleteManifestWork(mwName, mwNamespace)
+		if err != nil {
+			return fmt.Errorf("%w", err)
+		}
+
+		// check if the Spec has deleteOption. If not call createOrUpdate() to get
```

Comment:
Move the entire check and update delete options before actual deletion to a separate function, maybe into mw_util.go as part of the MWUtil interface, and invoke it here.

Any future namespace MW deletion can then ensure the delete options are set right and then the MW deleted, also simplifies code here.